### PR TITLE
Affixfile

### PR DESCRIPTION
**Pull request by Matthew-Mosior**
Hi everyone! 

I recently got WebtoHunspell.hs to take a single directory containing the various json paradigm files as the command line argument (runghc WebtoHunspell.hs /path/to/paradigm/directory/).  This definitely seems like the best way to go about it.  There is one caveat:  WebtoHunspell.hs can't be located in same directory as the paradigm json files.
 
I did some major re-factoring of the regular expression field for sub-rules, and implemented the ability to handle words where stripping occurs (i.e. teach -> taught).

As well as the major re-factoring, WebtoHunspell.hs is now generating out out.aff and out.dic pairs.

Matt

### CODE REVIEW

File: WebtoHunspell/WebtoHunspell.hs [hunk 1]
```diff
@@ -0,0 +1,1209 @@
...
+
+{-General Utility Functions.-}
+
+--linefeed -> To feed file into nested list line by line, 
+--delimited by newline characters.
+linefeed :: String -> [String]
+linefeed xs = L.lines xs
+
+--singleunnest
+singleunnest :: [a] -> a
```
Comment (pannapat) [inline WebtoHunspell/WebtoHunspell.hs:119]: Sorry but what dos the "unnest" and "mtriple" stand for?

File: WebtoHunspell/WebtoHunspell.hs [hunk 2]
```diff
@@ -0,0 +1,1209 @@
...
+maptupletotriplet _  []        = []
+maptupletotriplet x ((y,z):xs) = [tupletotriplet x (y,z)] ++ (maptupletotriplet x xs)
+
+{----------------------------}
+
+{-Functions to grab information from parsed JSON.-}
+
+--grabparadigm -> To grab the paradigm of the json.
+grabparadigm :: Maybe Paradigm -> String
+grabparadigm  Nothing                                  = []
```
Comment (pannapat) [inline WebtoHunspell/WebtoHunspell.hs:162]: Could you help explain what does this line do? Is it for null data handling?

**Review by pannapat — COMMENTED** [review]
Hi Matt, most of my comments are questions probably because I just learned the language. I could only review about 1/3 of the whole code and will try to review the rest asap.

**Comment by pannapat** [comment]
@Matthew-Mosior @Jyckle I'm just curious why this's always the entire file changed instead of some read and greens lines in this part of code. It's kind of difficult to focus which part of code has been changed. Or it's actually been changed the whole file?

File: WebtoHunspell/WebtoHunspell.hs (hunk 1 above)
Comment (Matthew-Mosior) [reply WebtoHunspell/WebtoHunspell.hs:119]: These are names of functions.  I tried to convey the functionality contained within the function in its name.

File: WebtoHunspell/WebtoHunspell.hs (hunk 2 above)
Comment (Matthew-Mosior) [reply WebtoHunspell/WebtoHunspell.hs:162]: Great question!  This function uses the Maybe monad.  This basically allows for a function to either return some value of type x, or return nothing in the case it isn't found.

**Review by pannapat — APPROVED** [review]
We agreed to merge this during our meeting.

**Pull request merged**